pkg/controller/podautoscaler/horizontal.go

```diff
@@ -953,7 +987,10 @@ func (a *HorizontalController) convertDesiredReplicasWithBehaviorRate(args Norma
 			return maximumAllowedReplicas, possibleLimitingReason, possibleLimitingMessage
 		}
 	} else if args.DesiredReplicas < args.CurrentReplicas {
+		a.scaleDownEventsLock.RLock()
 		scaleDownLimit := calculateScaleDownLimitWithBehaviors(args.CurrentReplicas, a.scaleUpEvents[args.Key], a.scaleDownEvents[args.Key], args.ScaleDownBehavior)
```

Comment:
this is accessing both scaleUpEvents and scaleDownEvents, but is only locking scaleDownEventsLock

that means our test coverage for concurrency safety is lacking

Reply:
good catch, thanks. `scaleUpEvents` has been added here later in another PR and I missed it during a rebase.